plugin/src/main/java/org/wildfly/plugin/provision/PackageServerMojo.java

```diff
@@ -318,15 +318,26 @@ private void warnExtraConfig(Path extraContentDir) {
         }
     }
 
-    private Path getDeploymentContent() {
+    private Path getDeploymentContent() throws MojoExecutionException {
         final PackageType packageType = PackageType.resolve(project);
         final String filename;
         if (this.filename == null) {
             filename = String.format("%s.%s", project.getBuild().getFinalName(), packageType.getFileExtension());
         } else {
             filename = this.filename;
         }
-        return targetDir.toPath().resolve(filename);
+        Path deployment = targetDir.toPath().resolve(filename);
+        if (!Files.exists(deployment)) {
```

Comment:
Very minor, but you could uses `Files.notExists(deployment)`. It just makes it slightly more readable.

Reply:
Thanks James, fixed.